homeassistant/helpers/device_registry.py

```diff
@@ -20,12 +20,13 @@
 class DeviceEntry:
     """Device Registry Entry."""
 
-    identifiers = attr.ib(type=list)
+    connections = attr.ib(type=set, converter=set)
+    identifiers = attr.ib(type=set, converter=set)
     manufacturer = attr.ib(type=str)
     model = attr.ib(type=str)
-    connection = attr.ib(type=list)
     name = attr.ib(type=str, default=None)
     sw_version = attr.ib(type=str, default=None)
+    config_entries = attr.ib(type=set, default=set(), converter=set)
```

Comment:
I think default should be a function, not a set?

Reply:
A method that creates a set and handles None or what are you thinking @balloob ?

Reply:
I will add a factory